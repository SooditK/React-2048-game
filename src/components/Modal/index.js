import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './modal.scss';

// Modal, parent componet for popup elements
export default function Modal({ display, children }) {
  return (
    <div className={classnames({ [styles.hidden]: !display })}>
      <div className={styles.overlay} />
      <div className={`${styles.modal} ${styles.center}`}>{children}</div>
    </div>
  );
}

Modal.propTypes = {
  display: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
};
