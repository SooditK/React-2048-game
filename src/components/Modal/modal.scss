@import '../../assets/styles/vars';

.hidden {
  display: none;
}

.overlay {
  background-color: rgba(0, 0, 0, 0.6);
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 10;
}

.modal {
  background-color: transparent;
  left: 50%;
  margin-left: -15vw;
  margin-top: -10vw;
  position: fixed;
  top: 50%;
  z-index: 11; /* 1px higher than the overlay layer */
}

@media #{$mobile} {
  .modal {
    margin-left: -37.5vw;
    margin-top: -15vw;
  }
}

.center {
  align-content: center;
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
