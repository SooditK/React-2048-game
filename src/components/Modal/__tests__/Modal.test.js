import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '..';

describe('<Modal />', () => {
  it('component render', () => {
    const modal = renderer
      .create(
        <Modal display>
          <div>test</div>
        </Modal>
      )
      .toJSON();
    expect(modal).toMatchSnapshot();
  });
});
