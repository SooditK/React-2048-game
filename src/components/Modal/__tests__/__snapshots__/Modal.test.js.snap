// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> component render 1`] = `
<div
  className=""
>
  <div
    className="overlay"
  />
  <div
    className="modal center"
  >
    <div>
      test
    </div>
  </div>
</div>
`;
